x/hub-genesis/keeper/ibc_module.go

```diff
@@ -49,8 +49,6 @@ func (w IBCModule) OnChanOpenConfirm(
 
 	state := w.k.GetState(ctx)
 	if state.CanonicalHubTransferChannelHasBeenSet() {
-		// We only set the canonical channel in this function, so if it's already been set, we don't need
-		// to send the transfers again.
 		return nil
 	}
 
```

Comment:
If im not missing anything I think there is a fundemntal problem with calling SubmitGenesisBridgeData while another channel is pending which is that you don't have enough money in the module to escrow again..
so you must have only 1 pending channel at a time.
which than leads to the thought that you can actually have only one channel at all, cause if the previous channel is still pending it will either succeed or retry and in either case no other channel opened can run this logic as the funds are already escrowed. 

timeouts here also doesn't help cause they will simply cause another retry.

i.e the problem is race condition on locking funds. Maybe a possible solution is to put it in a "shared" escrow and only really transfer it to the correct channel after success and only escrow it to the "shared" address on first attempt.